maindataclasses.rb

```diff
@@ -56,19 +69,37 @@
 end
 
 # task 7
+puts "task7"
 ary3 = [1,2,3,4,5,6,7,8]
 z = 1
 l = ary3.length
-while z > 0 && z < (l-1)
+while z < (l-1)
   if ary3[z].even?
-   ary3[z] += ary3[0]
+    ary3[z] += ary3[0]
   end
   z += 1
 end
 puts ary3
 
 # task 8
-hash = {'key1':'value1', 'key2':'value2'}
-hash.each_pair do |key,value|
-  puts hash.keys
-end
\ No newline at end of file
+puts "task8"
+hash = {'keyone' => 'value', 'keytwo' => 'value'}
+hash.each_key {|key| key.to_i}
```

Comment:
each_key and each_value methods don't modify initial values
these are just loop methods